File: src/main/java/com/ndgndg91/ordermatchingengine/order/event/OrderMatchTriggerEvent.java
package com.ndgndg91.ordermatchingengine.order.event;

import com.ndgndg91.ordermatchingengine.order.OrderType;
import com.ndgndg91.ordermatchingengine.order.PriceType;
import com.ndgndg91.ordermatchingengine.order.Symbol;
import lombok.Getter;
import lombok.ToString;

import java.time.LocalDateTime;

@Getter
@ToString
public final class OrderMatchTriggerEvent {
    private final Symbol symbol;
    private final PriceType priceType;
    private final OrderType orderType;
    private final LocalDateTime creationTime;

    public OrderMatchTriggerEvent(Symbol symbol, PriceType priceType, OrderType orderType) {
        this.symbol = symbol;
        this.priceType = priceType;
        this.orderType = orderType;
        this.creationTime = LocalDateTime.now();
    }
}

File: legacy/OrderMatchingEngineKotlin/src/main/kotlin/com/ndgndg91/ordermatchingenginekotlin/order/event/OrderMatchTriggerEvent.kt
package com.ndgndg91.ordermatchingenginekotlin.order.event

import com.ndgndg91.ordermatchingenginekotlin.order.OrderType
import com.ndgndg91.ordermatchingenginekotlin.order.PriceType
import com.ndgndg91.ordermatchingenginekotlin.order.Symbol
import java.time.LocalDateTime

data class OrderMatchTriggerEvent(
    val symbol: Symbol,
    val priceType: PriceType,
    val orderType: OrderType,
    val creationTime: LocalDateTime = LocalDateTime.now()
)
